cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -282,6 +285,60 @@ def validate_and_update_from_json(cls, block, jsondict, user, filter_tabs=True):
 
         return did_validate, errors, updated_data
 
+    @staticmethod
+    def get_dynamic_user_partition_id(block, min_partition_id, max_partition_id):
+        """
+        Get a dynamic partition id that is not already in use.
+        """
+        # Importing here to avoid ImportError:
+        # ImportError: cannot import name 'GroupConfiguration' from partially initialized module
+        # 'cms.djangoapps.contentstore.course_group_config' (most likely due to a circular import)
+        from cms.djangoapps.contentstore.course_group_config import GroupConfiguration
+        return generate_int_id(
+            min_partition_id,
+            max_partition_id,
+            GroupConfiguration.get_used_ids(block),
+        )
+
+    @classmethod
+    def fill_teams_user_partitions_ids(cls, block, settings_dict):
+        """
+        Fill the `dynamic_user_partition_id` in the team settings if it is not set.
+
+        This is used by the Dynamic Team Partition Generator to create the dynamic user partitions
+        based on the team-sets defined in the course.
+        """
+        if not CONTENT_GROUPS_FOR_TEAMS.is_enabled(block.id):
+            return
+
+        proposed_topics = cls.get_team_sets_from_settings(settings_dict)
+        for proposed_topic in proposed_topics:
+            if not proposed_topic.get('dynamic_user_partition_id'):
+                proposed_topic['dynamic_user_partition_id'] = cls.get_dynamic_user_partition_id(
+                    block,
+                    MINIMUM_STATIC_PARTITION_ID,
```

Comment:
An unfortunate side-effect of the this implementation is that the constant name `MINIMUM_STATIC_PARTITION_ID` no longer makes sense, as we now use it for _dynamic_ teamset partitions as well. The words "static" and "dynamic" are literally opposite 😆 😭 

Could you either rename it to something less confusing like `MINIMUM_UNUSED_PARTITION_ID` , or open an edx-platform backlog issue for renaming it?

Reply:
I opened a new small PR: https://github.com/openedx/edx-platform/pull/34529

Reply:
awesome, ty, I'll review that one right after this one